### PR TITLE
Create a CLI which will initialize a directory with a config file.

### PR DESCRIPTION
**Pull request by SamirTalwar**
### What

This will initialize the current directory with a single file, `configuration.json`, with an empty configuration.

If the directory contains any files, the command will refuse to run, as a precaution.

### How

It wraps the `configuration::empty` function.

### CODE REVIEW

File: crates/cli/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+use std::fs;
```
Comment (soupi) [inline crates/cli/src/lib.rs:1]: please docstrings on modules types and functions they really help me more than you think.

Comment (SamirTalwar) [reply crates/cli/src/lib.rs:1]: Oh, thank you for the reminder! I will add lots of comments.

Comment (SamirTalwar) [reply crates/cli/src/lib.rs:6]: Done.

File: crates/cli/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+//! The interpretation of the commands that the CLI cah handle.
+//!
+//! The CLI can do a few things. This provides a central point where those things are routed and
+//! then done, making it easier to test this crate deterministically.
```
Comment (soupi) [inline crates/cli/src/lib.rs:4]:
```suggestion
//! The interpretation of the commands that the CLI can handle.
//!
//! The CLI can do a few things. This provides a central point where those things are routed and
//! then done, making it easier to test this crate deterministically.
```

What is the purpose of the CLI? Is it only to initialize the config?

**Review by plcplc — APPROVED** [review]
And we're off to the races!

File: crates/cli/src/lib.rs (hunk 2 above)
Comment (SamirTalwar) [reply crates/cli/src/lib.rs]: Yes, for now. I added a bit more commentary to `main.rs`.

File: crates/cli/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+/// Run a command in a given directory.
+pub fn run(command: Command, configuration_dir: &Path) -> anyhow::Result<()> {
+    match command {
+        Command::Initialize => initialize(configuration_dir)?,
+    };
+    Ok(())
+}
+
+/// Initialize an empty directory with the connector configuration.
```
[the remark this answers is not in the document]
Comment (SamirTalwar) [reply crates/cli/src/lib.rs]: Stolen; thanks!

**Pull request merged**